tsl/src/fdw/deparse.c

```diff
@@ -867,6 +867,59 @@ deparseDistinctClause(StringInfo buf, deparse_expr_cxt *context)
 			return;
 	}
 
+	if (query->hasDistinctOn)
+	{
+		/*
+		 * Pushing down DISTINCT ON is more complex than plain DISTINCT.
+		 * The former requires a matching ORDER BY clause, or else the result is
+		 * undefined. There is a diagnostic for the case where the ORDER BY
```

Comment:
Hmm DISTINCT ON without ORDER BY has the ordering undefined like any other query without ORDER BY

Reply:
I reworded the comment.